### PR TITLE
Testing tox

### PR DESCRIPTION
**Pull request by caseylitton**
Updates for running tox tests on a local devstack.  Checking if they work in Travis

### CODE REVIEW

**Comment by caseylitton** [comment]
@stvstnfrd @kluo @caesar2164 PR to fix travis tests on Kevin's last commit.  The updates were based 
https://github.com/Stanford-Online/xblock-in-video-quiz 
but needed some tweaks.  This setup also passes local tox testing on a devstack.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+git+https://github.com/edx/XBlock.git#egg=XBlock
```
Comment (caesar2164) [inline requirements.txt:2]: enum34
django

Comment (caesar2164) [reply requirements.txt:2]: put those above this line...

File: tox.ini [hunk 2]
```diff
@@ -8,8 +8,6 @@ commands = {envpython} manage.py test
 [testenv:py27-dj18]
 deps =
     -rrequirements.txt
-    enum34
-    django >= 1.8, < 1.9
```
Comment (caesar2164) [inline tox.ini]: this line should stay

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+django<1.9,>=1.8
```
Comment (caesar2164) [inline requirements.txt:1]: this should just be "django"

Comment (stvstnfrd) [reply requirements.txt:1]: Should it? That means it will install the newest version.

Comment (caesar2164) [reply requirements.txt:1]: yeah, cause we don't care which version of django it uses except for the specific DJ1.8 test env

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,44 @@
+[tox]
+downloadcache = {toxworkdir}/_download/
+envlist = py27-dj18,coverage,pep8,pylint
+
+[testenv]
+commands = {envpython} manage.py test
+
+[testenv:py27-dj18]
+deps =
+    -rrequirements.txt
```
Comment (caesar2164) [inline tox.ini:10]: there should be a specific `django<1.9,>=1.8` here

File: requirements.txt (hunk 3 above)
Comment (caseylitton) [reply requirements.txt:1]: 1. In the tox.ini in three places 
Or 
2. Once in requirements and then gets installed for pep8 and pylint when it's not needed.

(1) is more efficient for loading and (2) is more efficient for running.  I'd opt to put it in the needed test in the tox.ini

File: requirements.txt (hunk 1 above)
Comment (caseylitton) [reply requirements.txt:2]: django was moved in from tox.ini and enum34 was add to setup.py.  Adding django to setup.py did not help.

File: tox.ini (hunk 4 above)
Comment (caseylitton) [reply tox.ini:10]: 1. In the tox.ini in three places 
Or 
2. Once in requirements and then gets installed for pep8 and pylint when it's not needed.

(1) is more efficient for loading and (2) is more efficient for running.  I'd opt to put it in the needed test in the tox.ini

File: freetextresponse/freetextresponse.py [hunk 5]
```diff
@@ -405,6 +407,7 @@ def _compute_score(self):
         )
 
     def _determine_credit(self):
+        # pylint: disable=redefined-variable-type
```
Comment (caesar2164) [inline freetextresponse/freetextresponse.py:412]: this seems like a failure we could fix, instead of just disabling it

**Comment by caesar2164** [comment]
this looks really good to me, I'd love some background on the disabled pylint tests tho...

File: freetextresponse/freetextresponse.py (hunk 5 above)
Comment (caseylitton) [reply freetextresponse/freetextresponse.py:412]: This is an issue with how pylint handles enums/conditionals.   I think it is fixed in a newer version.

Redefinition of result type from freetextresponse.freetextresponse.Credit.zero to freetextresponse.freetextresponse.Credit.full

Comment (caseylitton) [reply freetextresponse/freetextresponse.py:412]: This is still an issue in pylint 1.6.4

File: freetextresponse/tests.py [hunk 6]
```diff
@@ -3,11 +3,14 @@
 """
 import unittest
 
-import mock
 from django.test.client import Client
 from django.utils.translation import ugettext as _
+
+import mock
 from mock import MagicMock
```
Comment (dcadams) [inline freetextresponse/tests.py]: Do you need both of these imports?

File: setup.py [hunk 7]
```diff
@@ -1,24 +1,50 @@
 import json
 import setuptools
+from setuptools.command.test import test as TestCommand
```
Comment (dcadams) [inline setup.py:3]: Do you need both of these imports?

**Comment by dcadams** [comment]
:+1:

**Comment by caesar2164** [comment]
👍

**Pull request merged**